tools/tpm2_nvdefine.c

```diff
@@ -271,12 +294,17 @@ static tool_rc handle_no_index_specified(ESYS_CONTEXT *ectx, TPM2_NV_INDEX *chos
     return rc;
 }
 
-static tool_rc validate_size(void) {
+static tool_rc validate_size(ESYS_CONTEXT *ectx) {
+
+    tool_rc rc = tool_rc_general_error;
 
     switch ((ctx.nv_attribute & TPMA_NV_TPM2_NT_MASK) >> TPMA_NV_TPM2_NT_SHIFT) {
         case TPM2_NT_ORDINARY:
             if (!ctx.size_set) {
-                ctx.size = TPM2_MAX_NV_BUFFER_SIZE;
+		if ((rc = get_max_nv_index(ectx, &ctx.size)) != tool_rc_success) {
+			ctx.size = TPM2_MAX_NV_BUFFER_SIZE;
+			return rc;
```

Comment:
So fail validate_size but set the ctx.size to max? I don't even think that will work as callers of validate_size will cause the tool to exit. I think you want to return tool_rc_success